File: Platforms/Windows/Reign.Video.D3D9/SamplerState.cpp
#include "pch.h"
#include "SamplerState.h"

namespace Reign
{namespace Video
{namespace D3D9
{
	#pragma region Constructor
	SamplerState::SamplerState(DisposableI^ parent, SamplerStateDescI^ desc)
	: Disposable(parent)
	{
		video = parent->FindParentOrSelfWithException<Video^>();
		this->desc = (SamplerStateDesc^)desc;
	}

	SamplerState::~SamplerState()
	{
		disposeChilderen();
	}
	#pragma endregion
	
	#pragma region Methods
	void SamplerState::Enable(int index)
	{
		video->Device->SetSamplerState(index, D3DSAMP_MINFILTER, desc->Filter);
		video->Device->SetSamplerState(index, D3DSAMP_MAGFILTER, desc->Filter);
		video->Device->SetSamplerState(index, D3DSAMP_MIPFILTER, desc->Filter);

		video->Device->SetSamplerState(index, D3DSAMP_ADDRESSU, desc->AddressU);
		video->Device->SetSamplerState(index, D3DSAMP_ADDRESSV, desc->AddressV);
		video->Device->SetSamplerState(index, D3DSAMP_ADDRESSW, desc->AddressW);

		video->Device->SetSamplerState(index, D3DSAMP_BORDERCOLOR, desc->BorderColor);
	}
	#pragma endregion
}
}
}

File: Platforms/Windows/Reign.Video.D3D9/States/SamplerState.cs
using System;
using Reign.Core;
using Reign_Video_D3D9_Component;

namespace Reign.Video.D3D9
{
	public class SamplerState : Disposable, SamplerStateI
	{
		#region Properties
		private SamplerStateCom com;
		#endregion

		#region Constructors
		public static SamplerState New(DisposableI parent, SamplerStateDescI desc)
		{
			return new SamplerState(parent, desc);
		}

		public SamplerState(DisposableI parent, SamplerStateDescI desc)
		: base(parent)
		{
			var video = parent.FindParentOrSelfWithException<Video>();
			com = new SamplerStateCom(video.com, ((SamplerStateDesc)desc).com);
		}
		#endregion

		#region Methods
		public void Enable(int index)
		{
			com.Enable(index);
		}
		#endregion
	}
}